plugins/woocommerce/includes/wc-stock-functions.php

```diff
@@ -261,6 +273,44 @@ function wc_trigger_stock_change_notifications( $order, $changes ) {
 	$order->add_order_note( __( 'Stock levels reduced:', 'woocommerce' ) . ' ' . implode( ', ', $order_notes ) );
 }
 
+/**
+ * Check if a product's stock quantity has reached certain thresholds and trigger appropriate actions.
+ *
+ * This functionality was moved out of `wc_trigger_stock_change_notifications` in order to decouple it from orders,
+ * since stock quantity can also be updated in other ways.
+ *
+ * @param WC_Product $product        The product whose stock level has changed.
+ * @param int|float  $stock_quantity The new quantity of stock.
+ *
+ * @return void
+ */
+function wc_trigger_stock_change_actions( $product, $stock_quantity ) {
```

Comment:
Since we extracted it out, it would be great if we cover this flow with tests from both REST API and UI, if not already present.

Reply:
Fair point. I'll look into it

Reply:
Tests added in 3712135482